src/org/freedesktop/gstreamer/Gst.java

```diff
@@ -429,29 +474,57 @@ public static final void init() throws GstException {
      * <b>Note:</b><p>
      * This method will throw a GstException if it fails.
      * 
+     * @param requestedVersion the minimum requested GStreamer version.
      * @param progname the java program name.
      * @param args the java argument list.
      * @return the list of arguments with any gstreamer specific options stripped 
      * out.
      * @throws org.freedesktop.gstreamer.GstException
      */
-    public static synchronized final String[] init(String progname, String[] args) throws GstException {
+    public static synchronized final String[] init(Version requestedVersion,
+            String progname, String ... args) throws GstException {
+                  
+        if (CHECK_VERSIONS) {
+            Version availableVersion = getVersion();
+            if (requestedVersion.getMajor() != 1 || availableVersion.getMajor() != 1) {
+                throw new GstException("gst1-java-core only supports GStreamer 1.x");
+            }
+            if (requestedVersion.getMinor() < 8) {
+                requestedVersion = new Version(1, 8);
+            }
+            if (!availableVersion.checkSatisfies(requestedVersion)) {
+                throw new GstException("The requested version of GStreamer is not available\n"
```

Comment:
I would prefer String.format here that makes it more readable.

Reply:
Yes, good call!

Reply:
Done!